Yank/yank.py

```diff
@@ -431,7 +460,7 @@ def _create_phase(self, thermodynamic_state, alchemical_phase):
             integrator = openmm.VerletIntegrator(1.0 * unit.femtosecond)
             context = openmm.Context(alchemical_system, integrator)
             context.setPositions(positions[0])
-            for alchemical_state in alchemical_states:
+            for index, alchemical_state in enumerate(alchemical_states):
```

Comment:
The index only appears in the `raise` statement, the original code actually never assigned something to `index`. It would have been a wash in the end since an exception got raised, but lets raise the one we actually want to.